media/js/privacy/data-preferences.es6.js

```diff
@@ -0,0 +1,85 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+const preferenceCookieID = 'moz-1st-party-data-opt-out';
+const prefStatus = document.querySelector('.data-preference-status');
+const statusText = prefStatus.querySelector('.data-preference-text');
+let optOutText;
+let optInText;
+
+function updateStatus() {
+    if (hasOptedOut()) {
+        statusText.innerText = optOutText;
+        prefStatus.classList.add('is-opted-out');
+        prefStatus.classList.remove('is-opted-in');
+    } else {
+        statusText.innerText = optInText;
+        prefStatus.classList.add('is-opted-in');
+        prefStatus.classList.remove('is-opted-out');
+    }
+}
+
+function doOptOut() {
```

Comment:
**issue/question (non-blocking):** we don't prevent redundant clicks (i.e. opting out if you're already opted out, or opting in if you're already opted in), is it worth doing the `hasOptedIn` check here and in `doOptIn` to conditionally run the status updates only when there's a change? Is the idea with current implementation to allow folks to re-opt-out before the cookie expiry to push the expiry date further into the future?

Reply:
Good suggestion, I'll add the check here 👍